### ISSUE TITLE
better platform.txt subject headings

### ISSUE THREAD

**Issue by ag88** [rogerclarkmelbourne/Arduino_STM32#761]
in stm32duino.com forum, there are often users who installed both official cores and libmaple core.
and they are often confused about which core is it while reporting issues.

i'd like to suggest we add the word libmaple, in the _platform.txt_ name define so that they appear in the Arduino IDE boards subject headings 
https://github.com/rogerclarkmelbourne/Arduino_STM32/blob/master/STM32F1/platform.txt#L6
https://github.com/rogerclarkmelbourne/Arduino_STM32/blob/master/STM32F4/platform.txt#L6

**Comment by stevstrong** [comment]
Roger, what is your opinion on this?

**Comment by rogerclarkmelbourne** [comment]
Umm.

Since I have the stm32duino.com domain to STM (free of charge), and STM have been using that name as a trademark even prior to me giving them the domain, I agree it should be changed.

However, I'm not 100% sure what the the name could be 

Possibly just Arduino_STM32  to match this repo ?

I'm not sure everyone will know what "libmaple" means, because its a long time since LeafLabs abandoned this core.

**Comment by stevstrong** [comment]
Maybe (Arduino_STM32 aka Libmaple) ?

**Comment by rogerclarkmelbourne** [comment]
Perhaps 

Libmaple/ArduinoSTM32

**Comment by stevstrong** [comment]
That is promising :)

**Comment by rogerclarkmelbourne** [comment]
@ag88 

Comments?

**Comment by zoomx** [comment]
LibmapleRCM/ArduinoSTM32
just to be sure not be confused with the original Libmaple?

**Comment by ag88** [comment]
Hi Roger, all,Thanks for your response.You can decide on a name you feel comfortable with. My main motivation is simply due to that new users who installed both the official core and this sees stm32duino.com on the board menus in the Arduino IDE. Hence, they confuse this with the official STM core when they post their questions in the forum. As in that when we asked the new user what core are they using, they often reply to that they don't know and yet they are asking questions about the STM32duino cores.Hence, it suffices if we can distinguish this core in the board menus, by asking the user what do they see on the menu. E.g. if they see Arduino_STM32 it is, this core, otherwise it is not. I've originally suggested libmaple as this core is based around libmaple from leaflabs.

**Comment by stevstrong** [comment]
I would stick to Roger's suggestion, I think (Libmaple/ArduinoSTM32) would describe correctly the core.

**Comment by fpistm** [comment]
From my point of view, stm32duino means stm32 support in Arduino.
stm32duino Libmaple or something like that.
Why not also mention Roger as it relies on your GH username?

**Comment by rogerclarkmelbourne** [comment]
It doesn't need to have any mention of me in the name.

I think RCM etc, would just be confusing.

I just did a google search for "libmaple" and was very surprised to find that the original Leaflabs repo , is still top of the list, as are the next 3 hits.

So, this seems to indicate that not many people realise this repo is the active derivative of LibMaple

And perhaps even including LibMaple in the name would be confusing.

As the big difference between this repo and the STM32 official core, is that it doesn't use the HAL, perhaps this could be included in the name. But then again, perhaps hardly anyone would understand "non-HAL"or some similar text


BTW. I never expected people to still be using this core several years after the HAL core was available.

**Comment by stevstrong** [comment]
Yes, it seems Libmaple repository to be the first result, but it is arked as retired.
So I think mentioning Libmaple here would not be wrong.
Maybe then (Libmaple-based)?
Or simply (Arduino_STM32)?

**Comment by fpistm** [comment]
Don't know if you used the latest Arduino IDE, this is how the menu is displayed now:
![image](https://user-images.githubusercontent.com/20641798/96098878-ea320600-0ed2-11eb-8884-fa232ab52047.png)

Note that like tools directories have a boards.txt they are displayed.

**Comment by ag88** [comment]
In a sense we have maintained and enhanced libmaple core introducing various fixes and improvements. To the credit of the origins, I'd think it is appropriate to mention this as a libmaple core.
I kind of remember one of the fixes for uart I added is to drop bytes with corrupted stop bits or parity errors. This makes various sketches that expects a clean connection work more reliably. Many others have contributed other fixes.
